lib/httpServer/transports/http/index.js

```diff
@@ -615,6 +638,57 @@ function safeDecodePath(path) {
 	}
 }
 
+/**
+ * Encapsulate the request object
+ *
+ * This function currently only takes care of
+ * checking the HTTP POST size.
+ *
+ * @param {*} req HTTP request
+ */
+function setRequestFilters(req) {
+	var realOn = req.on.bind(req);
+	var byteSize = 0;
+
+	function isRequestTooLarge() {
+		return byteSize > maxRequestSize;
+	}
+
+	req.on('data', function (chunk) {
+		byteSize += chunk.length;
+
+		if (isRequestTooLarge()) {
+			req.emit('error', new RequestError('Entity too large', req));
+			req.removeAllListeners();
+		}
+	});
+
+	// Here, we would normally want to use streams, but unfortunately streams
+	// do not wrap or proxy methods from upstream objects; therefore, we need to
+	// return the request object as-is, but still provide a mechanism to interrupt
+	// that allows us to interrupt the emission of data event in the case
+	// where the request was already found to be too large.
+	//
+	// The other reason we need this is because emitting an error event from within
+	// an event callback will not interrupt the current event emission; for instance, if
+	// you have two callbacks listening on the 'data' event and the first one emits an error,
+	// the second callback will still get emitted. removeAllListeners also do not have
+	// any effect in this case.
+	req.on = function (eventName, func) {
+		if (eventName !== 'data') {
+			return realOn(eventName, func);
+		}
+
+		realOn('data', function (chunk) {
+			if (!isRequestTooLarge()) {
+				func(chunk);
+			}
+		});
+	};
+
+	return req;
```

Comment:
Returning `req` makes it seem (from the call-site) that you may not get back what you put in. But you do. Perhaps not return anything?

Reply:
Should we change the behavior of this function in the future, we'd then need to change the call site as well. I prefer not relying explicitly on the existence of side-effects in general.